mlflow/tracking/fluent.py

```diff
@@ -3053,32 +3053,67 @@ def setup_autologging(module):
             register_post_import_hook(setup_autologging, "pyspark.ml", overwrite=True)
 
 
-_ACTIVE_MODEL_ID = ContextVar("active_model_id", default=None)
+class ActiveModelContext:
```

Comment:
Can we make this object immutable?

Reply:
I removed the setters. 
I didn't adopt dataclass(frozen=True) since in that case (something like this)
```
@dataclass(frozen=True)
class ActiveModelContext:
    model_id: Optional[str]=None
    set_by_user: bool = False

@property
def model_id(self):
    return self.model_id or MLFLOW_ACTIVE_MODEL_ID.get()
```
This property definition is not allowed because of its name conflicts with the attribute, and we need it to fetch the attribute easily. Let me know if you have better options :)

Reply:
The current impl LGTM :)